### DISCUSSION TITLE
Updating score for BM25 batch search

### DISCUSSION THREAD

**Discussion by Fatima-Haouari** [castorini/pyserini#1166] [Q&A] [answered]
Hi,
I indexed my own collection and I am using BM25 batch search for retrieval.  If I want to update the BM25 scores such as multiplying the score by X before retrieving top N documents. Is there a way to achieve this?

**Comment by lintool** [comment, marked as answer]
Can you explain the use case? You can access Lucene queries directly in Pyserini as follows:
https://github.com/castorini/pyserini/blob/master/docs/usage-querybuilder.md

**Reply by Fatima-Haouari** [reply]
What I achieved is retrieving top 1000 using BM25 then multiplying the BM25 scores by X value relevant to my problem and reranked again the retrieved documents. What I am trying to achieve is applying my scoring function BM25 score*X while retrieving so the scoring is applied to all documents not only top 1000.

**Reply by lintool** [reply]
The standard solution is to retrieve top _K_ (> 1000) and rerank those hits to obtain the top 1000. The only other option is the write a custom Lucene `Similarity`.

**Reply by Fatima-Haouari** [reply]
Thanks a lot for your quick response.
